include/cmr_geometry_utils/basic_geometry_utils.hpp

```diff
@@ -7,17 +7,45 @@
 
 namespace cmr_geometry_utils {
 
-double compute_distance(const geometry_msgs::msg::Point & start, const geometry_msgs::msg::Point & goal)
+double compute_distance_2d(const geometry_msgs::msg::Point & start, const geometry_msgs::msg::Point & goal)
 {
   return std::hypot(goal.x - start.x, 
                     goal.y - start.y);
 }
 
-double compute_angle_distance(const geometry_msgs::msg::Quaternion & start, const geometry_msgs::msg::Quaternion & goal)
+double compute_distance_2d(const geometry_msgs::msg::Pose & start, const geometry_msgs::msg::Pose & goal)
+{
+  return std::hypot(goal.position.x - start.position.x, 
+                    goal.position.y - start.position.y);
+}
+
+double compute_distance_2d(const geometry_msgs::msg::PoseStamped & start, const geometry_msgs::msg::PoseStamped & goal)
+{
+  return std::hypot(goal.pose.position.x - start.pose.position.x, 
```

Comment:
perhaps just define this overload in terms of the prior one.  ie pass `goal.pose` and `start.pose` to the first `compute_distance_2d`